### PR TITLE
[New Rule] File Compressed or Archived into Common Format

### PR DESCRIPTION
**Pull request by brokensound77**
## Summary
Creates a new BBR to detect compressed and archived files based on file header. All existing compressions detections are based on process names and subsets of activity.

This also includes a new command to generate the threat map entry `build-threat-map-entry`.

### CODE REVIEW

File: rules_building_block/collection_common_compressed_archived_file.toml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+building_block_type = "default"
+description = """Detects files being compressed or archived into common formats. This is a common technique used to
+obfuscate files to evade detection or to staging data for exfiltration.
+"""
+from = "now-119m"
+interval = "60m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+max_signals = 50000
```
Comment (brokensound77) [inline rules_building_block/collection_common_compressed_archived_file.toml:20]: 2 hours worth of compressions can be quite a lot

**Review by terrancedejesus — APPROVED** [review]
Looks good!

File: detection_rules/schemas/definitions.py [hunk 2]
```diff
@@ -90,6 +90,7 @@
     'OS: Linux',
     'OS: macOS',
     'OS: Windows',
+    'Rule Type: BBR',
```
Comment (terrancedejesus) [inline detection_rules/schemas/definitions.py:93]: YES...not sure how this was missed.

**Review by w0rk3r — APPROVED** [review]
Just a nit and, LGTM 🧙🏼‍♂️ 🧙🏼‍♂️

**Pull request merged**